examples/scripts/direct_reconstruction.py

```diff
@@ -1,3 +1,19 @@
+# ---
```

Comment:
Having this part in every script is very error prone. 

Can we add it only during the translation process from py to ipynb without committing it?

Reply:
There is a lot of version information in there which would be a lot of work to keep up-to-date going forward.

Jupytext did not need this before but it was provided via command line arguments. Maybe we can try to get something similar working here. 

I would not encourage modifying notebooks and then update the scripts but it should always work as script -> notebook. Otherwise I still believe it is far too dangerous that people commit notebook content and make our repo grow big again.

Reply:
> There is a lot of version information in there which would be a lot of work to keep up-to-date going forward.

This version informations are only jupytext converter information which is now fixed in `.pre-commit-config.yaml`. The `kernelspec` information is set only as default value once and will not be changed again.

> Otherwise I still believe it is far too dangerous that people commit notebook content and make our repo grow big again.

We have a `nbstripout` in pre-commit which cleans outputs and any other metadata added by user executution of a notebook. This also happes **before** jupytext does any synchronization attemps. So there is no chance by now how the execution of the notebook by the user can affect the .ipynb and .py data.

I cannot see a scenario how to ruin the repo with this workaround.

The only changes in notebok cells will affect the .ipynb and .py files. So if any code line is changed either in .py or .ipynb it will be reflected in the paired representation. Isn't it more intuitive for a developer than:
1) changing the .py file
2) then execute the jupytext to get an updated .ipynb representation
3) run the .ipynb to verify the expected behaviour
4) if the behaviour is then expected -> go to 1.

Reply:
We cannot enforce that people run a pre-commit before committing so it is actually really easy to add the notebook content to the repo. 

We would need to enforce that people can only contribute via a fork but as you have experienced that does not work with our pipeline at the moment.

Reply:
> We cannot enforce that people run a pre-commit before committing so it is actually really easy to add the notebook content to the repo.

Okay, this makes sense. But in this case there is also no option to protect thre repo.

> that does not work with our pipeline at the moment.

@fzimmermann89 and me found the issue with the CI, I'll fix it soon.